File: backend/app/model/dto/response/userDto.py
from pydantic import BaseModel, Field
from typing import Optional
import uuid


class UserDto(BaseModel):
    id: uuid.UUID = Field(..., description="User unique identifier")
    email: str = Field(..., description="User email address")
    display_name: Optional[str] = Field(None, description="User display name")
    picture_url: Optional[str] = Field(None, description="User profile picture URL")
    sex: Optional[str] = Field(None, description="User sex")
    birth_year: Optional[int] = Field(None, description="User birth year")
    daily_step_goal: int = Field(10000, description="Daily step goal")
    timezone: str = Field("America/Lima", description="User timezone")

    class Config:
        from_attributes = True
        json_schema_extra = {
            "example": {
                "id": "123e4567-e89b-12d3-a456-426614174000",
                "email": "user@example.com",
                "display_name": "John Doe",
                "picture_url": "https://example.com/photo.jpg",
                "sex": "male",
                "birth_year": 1990,
                "daily_step_goal": 10000,
                "timezone": "America/Lima"
            }
        }

File: mobile/src/main/java/com/example/harbit/data/remote/dto/UserDto.kt
package com.example.harbit.data.remote.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class UserDto(
    @SerialName("id")
    val id: String, // UUID as string
    
    @SerialName("email")
    val email: String,
    
    @SerialName("display_name")
    val displayName: String?,

    @SerialName("preferred_email")
    val preferredEmail: String?,

    @SerialName("phone")
    val phone: String?,

    @SerialName("height")
    val height: Float?,

    @SerialName("weight")
    val weight: Float?,
    
    @SerialName("picture_url")
    val pictureUrl: String?,
    
    @SerialName("sex")
    val sex: String? = null,
    
    @SerialName("birth_year")
    val birthYear: Int? = null,
    
    @SerialName("daily_step_goal")
    val dailyStepGoal: Int = 10000,
    
    @SerialName("timezone")
    val timezone: String = "America/Lima"
)
